### PR TITLE
KOGITO-7057: [KSW][OpenAPI] - Add Token Propagation support to OpenAPI Extension

### PR DESCRIPTION
**Pull request by wmedvede**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributors guide](CONTRIBUTING.md)
- [X] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [X] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [X] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains link to any dependent or related Pull Request
- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

- <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by wmedvede** [comment]
Hi @ricardozanini, this PR bumps the quarkus-openapi-generator to the 0.7.0 version that includes the token propagation and adds a good set of tests to ensure that it works well also in the SW context.
Would you mind review?

Thanks.

**Comment by wmedvede** [comment]
jenkins rerun kogito-examples tests

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/ExternalServiceMock.java [hunk 1]
```diff
@@ -38,11 +39,13 @@ public class ExternalServiceMock implements QuarkusTestResourceLifecycleManager
...
     public static final String GENERATE_ERROR_QUERY = "GENERATE_ERROR_QUERY";
 
+    public static final String EXTERNAL_SERVICE_MOCK_URL = "external-service-mock.url";
+
     private WireMockServer wireMockServer;
 
     @Override
     public Map<String, String> start() {
-        wireMockServer = new WireMockServer(9292);
+        wireMockServer = new WireMockServer(options().dynamicPort());
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/ExternalServiceMock.java:48]: hmm I think we have quite a few tests using a fixed port. We could do the same in the rest of them.

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        // prepare the headers to pass to the token_propagation SW.
+        // service token-propagation-external-service1 and token-propagation-external-service2 will receive the AUTHORIZATION_TOKEN 
+        headers.put(HttpHeaders.AUTHORIZATION, AUTHORIZATION_TOKEN);
+        // service token-propagation-external-service3 will receive the SERVICE3_AUTHORIZATION_TOKEN
+        headers.put(SERVICE3_HEADER_TO_PROPAGATE, SERVICE3_AUTHORIZATION_TOKEN);
+        // service token-propagation-external-service4 will receive the SERVICE4_AUTHORIZATION_TOKEN
+        headers.put(SERVICE4_HEADER_TO_PROPAGATE, SERVICE4_AUTHORIZATION_TOKEN);
+
+        JsonPath jsonPath = newProcessInstance("/token_propagation", processInput, headers);
+        Assertions.assertThat(jsonPath.getString("id")).isNotBlank();
```
Comment (ricardozanini) [inline quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java:58]: Is it possible to verify if the token has propagated to the target service?

**Review by ricardozanini — APPROVED** [review]
LGTM! Great work!

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java (hunk 2 above)
Comment (wmedvede) [reply quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java:58]: @ricardozanini All the propagation use cases are verified in the method below, i.e. if any of the propagations didn't propagate the expected token for each case, the test will fail.
I think it's ok.

here https://github.com/wmedvede/kogito-runtimes/blob/ec112b3c46a9abd7fddd9c2ef1e6b7248fca2058/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationExternalServicesMock.java#L53

**Review by ricardozanini — APPROVED** [review]

**Comment by wmedvede** [comment]
jenkins retest this

**Pull request merged**